test/functional/logger.test.js

```diff
@@ -54,29 +54,24 @@ describe('Logger', function() {
   it('should not fail with undefined id', {
     metadata: { requires: { topology: ['single'] } },
 
-    test: function(done) {
-      var self = this;
-
+    test: function() {
       // set a custom logger per http://mongodb.github.io/node-mongodb-native/2.0/tutorials/logging/
       Logger.setCurrentLogger(function() {});
       Logger.setLevel('debug');
 
-      connectToDb('mongodb://localhost:27017/test', self.configuration.db, function(
-        err,
-        db,
-        client
-      ) {
-        expect(err).to.not.exist;
-
-        // perform any operation that gets logged
-        db.collection('foo').findOne({}, function(err) {
-          expect(err).to.not.exist;
+      return withClient(
```

Comment:
I don't think it'll drop the database after each test, since this isn't passing the `drop` helper option to `withDb`. 

My goal with these helpers was to move all of the startup/cleanup out of the test code, so that it is handled in a consistent manner and is more visible since it is now outside the test code.  That said I take your point, the `withDb` and `withCollection` helpers have kind of been generalized to the point of not adding value (they started out as much simpler helpers to drop the db/collection after a test).

Reply:
I went ahead and removed them.